app/(main)/(leads)/leads.tsx

```
@@ -119,14 +118,19 @@ const Leads = () => {
     const leadData = await getDocs(leadCols);
     const leads = leadData.docs.map((doc) => doc.data());
-
     setLeads(leads);
+    setFilteredLeads(leads);
   };
 
   useEffect(() => {
     fetchLeads();
-    setFilteredLeads(leads);
-  }, [leads]);
+  }, [, useIsFocused()]);
```

Biome (1.9.4)

[error] This array contains an empty slot.

Unsafe fix: Replace hole with undefined


(lint/suspicious/noSparseArray)